cma/evolution_strategy.py

```diff
@@ -4646,4 +4660,28 @@ def update(es):
     es.objective_function_complements = [_al]
     es.augmented_lagrangian = _al
     es.best_feasible = best_feasible_solution
+
+    if post_optimization:
+        positive_constraints = np.where(np.array(g(es.result.xfavorite)) > 0)
+        if len(positive_constraints[0]) > 0:
+            x_post_opt, es_post_opt = fmin_con(lambda x: sum([gi**2 if gi > 0 else 0 for gi in g(x)]),
+                                               es.result.xfavorite, sigma0 / 1000, g=g, h=h, **kwargs)
+            if es_post_opt.best_feasible.info is not None:
+                f_x_post_opt = objective_function(es_post_opt.best_feasible.info["x"])
+
+                best_feasible_solution_post_opt = ot.BestSolution2()
+                best_feasible_solution_post_opt.update(
+                    f_x_post_opt, info={
+                        'x': es_post_opt.best_feasible.info["x"],
+                        'f': f_x_post_opt,
+                        'g': es_post_opt.best_feasible.info["g"],
+                        'g_al': es_post_opt.best_feasible.info["g_al"]})
+                es.best_feasible_post_opt = best_feasible_solution_post_opt
+            else:
+                utils.print_warning('Post optimization was unsuccessful',
+                                    verbose=es.opts['verbose'])
+        else:
+            utils.print_warning('No positive constraint in ``es.results.xfavorite``, skipping post optimization',
+                                verbose=es.opts['verbose'])
+
     return es.result.xfavorite, es
```

Comment:
Returning `es.result.xfavorite` seems counterintuitive when `post_optimization is True`.

Reply:
I agree, I changed it so that the first return will be the best feasible solution found in post-optimization.